src/main/java/be/seeseemelk/mockbukkit/block/BlockMock.java

```diff
@@ -363,15 +369,14 @@ public boolean breakNaturally(ItemStack tool)
 	@Override
 	public BlockData getBlockData()
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		return blockData;
 	}
 
 	@Override
 	public void setBlockData(BlockData data)
 	{
-		// TODO Auto-generated method stub
-		throw new UnimplementedOperationException();
+		Assert.assertNotNull(data);
```

Comment:
Never assert in the implementations of normal methods. This will make any test fail which may actually intentionally provoke a null argument here.

You should always throw an `IllegalArgumentException` for illegal arguments instead of asserting this early on unless specifically intended.

Reply:
You could also use `Objects.requireNonNull(data)` instead